### PR TITLE
Support for math content in TeX-syntax for HTML and Markdown (Mistune)

### PR DESCRIPTION
**Pull request by hjhsalo**
* Add MathJax.js to render content in TeX-syntax
* Make Mistune (markdown parser) and Python-Markdown compatible with TeX-syntax (`$ ... $`, `$$ ... $$`, `\begin ... \end`). Otherwise characters matching markdown control characters (e.g. `__ ... __` or `** ... **`) that reside inside TeX-content, might mess up the TeX-content.

### CODE REVIEW

**Comment by hjhsalo** [comment]
Note that this doesn't necessarily close #1447 since I haven't implemented matching functionality in Python-Markdown. We haven't yet decided which one of the two parsers to use.

**Comment by ioannistsanaktsidis** [comment]
@hjhsalo is this one ready for merging ? If yes please rebase  and I will merge it.

**Comment by hjhsalo** [comment]
Reviewer can use `http://127.0.0.1/md/debug` as a test page (even without running `populate-instance.sh`) or http://127.0.0.1/record/328, which contains quite a lot of "real" math content (Dataset from the ATLAS Higgs Boson Machine Learning Challenge 2014). For comparing the results see http://opendata.cern.ch/record/328

**Comment by hjhsalo** [comment]
Comments and opinions about the MathJax config used currently are most welcome :)

**Comment by hjhsalo** [comment]
@ioannistsanaktsidis 
Should be ready for merge now (I haven't rebased it today), but it might be good to check the code e.g. for style issues, etc.

**Comment by ioannistsanaktsidis** [comment]
@hjhsalo  it works fine :) just one question, do we need mistune ? cc @tiborsimko  @pamfilos

File: cernopendata/templates/cernopendata_theme/page.html [hunk 1]
```diff
@@ -52,5 +52,22 @@
...
+        tex2jax: {
+          inlineMath: [ ['$','$'], ["\\(","\\)"] ],
+          displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
+          processEscapes: true
+        },
+        "HTML-CSS": { availableFonts: ["TeX"] }
+      });
+    </script>
+
+    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
```
Comment (ioannistsanaktsidis) [inline cernopendata/templates/cernopendata_theme/page.html]: Keep in mind this https://www.mathjax.org/cdn-shutting-down/ .

Comment (hjhsalo) [reply cernopendata/templates/cernopendata_theme/page.html]: Hmm, I've seen that I thought that I had taken it into account. I will check it once again.

Comment (hjhsalo) [reply cernopendata/templates/cernopendata_theme/page.html]: Did I misunderstood something?
The PR uses the recommended _cdnjs.cloudflare.com_ URL

> The easiest way for most site owners will be to simply switch to another CDN provider.
> 
> We recommend cdnjs which also uses CloudFlare for delivery (and on the higher “enterprise” level!). We have been in touch with cdnjs’s maintainers and will help push future MathJax releases to cdnjs.
> 
> For example, if you have been using the latest MathJax version (v2.7.0) change
> 
> <script type="text/javascript" async
>   src="https://cdn.mathjax.org/mathjax/2.7-latest/MathJax.js?...">
> </script>
> 
> to
> 
> <script type="text/javascript" async
>   src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?...">
> </script>
>

Comment (ioannistsanaktsidis) [reply cernopendata/templates/cernopendata_theme/page.html]: The comment was mostly for considering adding cdn as a bundle via npm and completely get rid of the cdn. Check this https://www.npmjs.com/package/mathjax . But it is ok for now we can leave it as it is. Just to keep in mind for future.

Comment (hjhsalo) [reply cernopendata/templates/cernopendata_theme/page.html]: OK. Thanks.

**Comment by hjhsalo** [comment]
> just one question, do we need mistune ?

Mistune provides better performance.

Python-Markdown provides better extensibility through extensions (e.g. proper and defined extension API / pattern, more extension available than for Mistune).

(I haven't made any benchmarking or comparing performance comparing between cod-mistune cod-markdown.)

I've written the included cod-mistune module to be feature-wise on par with the configuration (Python-Markdown extensions and their configuration) of cod-markdown-module.

Basically I wanted to keep Mistune around in case there are any performance issues with Python-Markdown. So far, none have been observed and probably won't be observed given that we rely on quite aggressive caching.

I would merge this one as is, and if cod-mistune needs to be removed let's remove it in a separate commit, just to keep cod-mistune in the code base in case someone find use for it in the future?
